### PR TITLE
Issue/11196 widget editor screen

### LINKED ISSUE #11196: Widget editor screen
Replicate the following UI used in Analytics hub:

Image

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #11196

### Description

Ads new screen for adding/removing or reordering the available widget from My Store
In order to avoid massive UI code duplication, I've refactored some existing components used in `AnalyticsHubSettingScreen` into more general-purpose ones. 

### Testing instructions

1. Run the app and click on the new `Settings` CTA on the top right corner
2. Select unselect items
3. Reorder them
4. Clickin on the `X` button and check the confirmation dialog is shown
5. Clickin on `Save` button will crash the app as it is not yet implemented

**Analytics hub settings**
Smoke test analytics hub settings screen to ensure everything works as expected and there are no regressions after the refactor

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/2e564e6e-a585-48c3-a0e3-dfb464cf2cc7

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>DashboardWidgetEditorViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>WidgetEditorState</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/204">18.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by 0nko — APPROVED** [review]
The changes look good and drag & drop functionality works as expected 👍. Nice work!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/DragAndDropList.kt [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import com.woocommerce.android.ui.compose.DraggableItem
+import com.woocommerce.android.ui.compose.component.ShowDividers.All
+import com.woocommerce.android.ui.compose.component.ShowDividers.Bottom
+import com.woocommerce.android.ui.compose.component.ShowDividers.None
+import com.woocommerce.android.ui.compose.dragContainerForDragHandle
+import com.woocommerce.android.ui.compose.rememberDragDropState
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun <T> DragAndDropItemsList(
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/DragAndDropList.kt]: Just one minor np comment before you merge this @JorgeMucientes, WDYT about making it clear that this is a selectable drag-and-drop list? If I were outside of this project, I'd think that this is just a plain drag-and-drop list.
Something like `DragAndDropSelectableItemsList`?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/widgeteditor/MyStoreWidgetEditorViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    ) : Parcelable
+
+    @Parcelize
+    data class MyStoreWidget(
+        val title: String,
+        val isSelected: Boolean,
+        val type: WidgetType,
+    ) : Parcelable
+
+    enum class WidgetType {
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/widgeteditor/MyStoreWidgetEditorViewModel.kt]: @0nko introduced a domain model to define the widgets in https://github.com/woocommerce/woocommerce-android/pull/11218, we should probably use it here (can be done in a separate PR).
I just wanted to bring this related discussion https://github.com/woocommerce/woocommerce-android/pull/11218#discussion_r1549494132 to your attention about the need of using a `sealed interface` for defining the widgets.

**Comment by JorgeMucientes** [comment]
Hey folks thank you for the review @0nko and @hichamboushaba. 
I've applied the suggested change for the drag and drop selectable list.
About the sealed interface model, cool, thanks for sharing the context. Yeah I think we can refactor that when connecting everything together. 

Additionally, I've applied some changes to the existing code in this PR to move all the changes I had added to work with new `Dashboard` fragment and viewModel replicas and leave the original MyStoreFragment files untouched.

**Pull request merged**